src/NovelCoreClrService.h

```diff
@@ -0,0 +1,54 @@
+//
+// Created by tagoo on 19/09/2019.
+//
+
+#ifndef NOVELRT_NOVELCORECLRSERVICE_H
+#define NOVELRT_NOVELCORECLRSERVICE_H
+
+#include <functional>
+#include <coreclr_delegates.h>
+#include <hostfxr.h>
+#include "Lazy.h"
+
+#if defined(_WIN32)
+  #define STR(s) L ## s
+  #define CH(c) L ## c
+#else
+  #define STR(s) s
+  #define CH(c) c
+#endif
+
+namespace NovelRT {
+  class NovelCoreClrService{
+  public:
+    NovelCoreClrService();
+    ~NovelCoreClrService();
+
+    template <class _Fty>
+    std::function<_Fty> getFunction(const char_t* assemblyPath, const char_t* typeName, const char_t* methodName, const char_t* delegateTypeName);
+
+  private:
+    Lazy<hostfxr_handle> _hostContextHandle;
+    Lazy<void*> _hostfxr;
+    Lazy<hostfxr_initialize_for_runtime_config_fn> _hostfxr_initialize_for_runtime_config;
+    Lazy<hostfxr_get_runtime_delegate_fn> _hostfxr_get_runtime_delegate;
+    Lazy<hostfxr_close_fn> _hostfxr_close;
+    Lazy<load_assembly_and_get_function_pointer_fn> _load_assembly_and_get_function_pointer;
+  };
+
+  template <class _Fty>
+  std::function<_Fty> NovelCoreClrService::getFunction(const char_t* assemblyPath, const char_t* typeName, const char_t* methodName, const char_t* delegateTypeName) {
```

Comment:
This could be made a bit "simpler" and have some strings be auto-constructed, etc.
However, this is zero overhead and most things (like the Ink service) will be their own APIs that can abstract this away.